### PR TITLE
Extract Configuration class

### PR DESCRIPTION
**Pull request by fatkodima**
Extraction for #442

### CODE REVIEW

File: lib/rack/attack.rb [hunk 1]
```diff
@@ -31,82 +32,8 @@ class MissingStoreError < Error; end
...
-        end
-      end
-
-      def tracked?(request)
-        tracks.each_value do |track|
-          track.matched_by?(request)
-        end
-      end
+      attr_accessor :enabled, :notifier
+      attr_reader :config
```
Comment (grzuy) [inline lib/rack/attack.rb]: Can we please use non abbreviated name, i.e. `configuration`?

File: lib/rack/attack.rb [hunk 2]
```diff
@@ -158,19 +76,16 @@ def call(env)
       env['PATH_INFO'] = PathNormalizer.normalize_path(env['PATH_INFO'])
       request = Rack::Attack::Request.new(env)
 
-      if safelisted?(request)
+      if @config.safelisted?(request)
```
Comment (grzuy) [inline lib/rack/attack.rb]: Can we please use the reader instead of the instance variable so that it's easier to detect typos?

**Comment by fatkodima** [comment]
Updated.

**Comment by grzuy** [comment]
:clap:

**Pull request merged**